File: make_it_white/white.cpp
#include <iostream>
#include <string>

using namespace std;

int main(void)
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n;
        string s;
        cin >> n >> s;
        bool found_first = false;
        int first = 0;
        int last = 0;
        for (int j = 0; j < n; j++)
        {
            if (s[j] == 'B' && found_first)
            {
                last = j;
            }

            else if (s[j] == 'B')
            {
                found_first = true;
                first = j - 1;
                last = j;
            }
        }

        cout << last - first << "\n";
    }
}

File: make_it_white/white.py
def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input()
        found_first = False
        first = 0
        last = 1
        for i in range(n):
            if s[i] == 'B' and found_first:
                last = i
            elif s[i] == 'B':
                found_first = True
                first = i - 1
                last = i
        print(last - first)
    
    
if __name__ == "__main__":
    main()
